### ISSUE TITLE
Does 'too long query locus' is RAM-dependent?

### ISSUE THREAD

**Issue by alejandrogzi** [hillerlab/TOGA#48]
Hi Michael and Bogdan!

After running some test in limited chromosomes, I run the official TOGA run with complete files. I noticed that SPLIT_CESAR.txt inside the rejected dir has some genes with the notation 'too long query locus'. Since the chain limit is correct but are not considered big_mem jobs, does this problem is ram-dependent? I will actually extend the cesar_chain_limit to capture all those transcripts that exceeded 100 chains, should I extend also the cesar_mem_limit to capture those 'too long query locus'? 

Best,
Alejandro

**Comment by kirilenkobm** [comment]
Hi @alejandrogzi,

Thank you for your question. I will definitely clarify the reasons why a gene might be omitted in the comprehensive TOGA documentation that is coming soon.

Regarding the "too_long_query_locus" reason for skipping transcript annotation:
It occurs when the following conditions are met:

Firstly, the subject chain is "spanning" with respect to the gene, meaning it is syntenic but has no aligning blocks within the gene body. In other words, the gene's exons and introns are not aligned, although the flanking regions are.
Then, one of the following conditions is required:
The respective orthologous locus (where the chain points to in the query genome) is 50 times longer than the gene body. Alternatively,
The respective orthologous locus is 1M bp longer than the gene body length.
(e.g., it's 1 AND (2 OR 3))

In summary, the reasons for implementing these filters are:

It is very unlikely that such query loci contain meaningful orthologous projections.
CESAR jobs with these characteristics would consume a significant amount of time and memory.

**Comment by alejandrogzi** [comment]
@kirilenkobm,

Thanks for your clear response. 

Alejandro
